tests/media/media.spec.ts

```diff
@@ -0,0 +1,28 @@
+import { media } from '@/media'
+import { isRef } from 'vue'
+
+describe('media', () => {
+    
+    beforeAll(() => {
+      Object.defineProperty(window, "matchMedia", {
+        writable: true,
+        value: jest.fn().mockImplementation(query => ({
+          matches: false,
+          media: query,
+          onchange: null,
+          addListener: jest.fn(), // Deprecated
+          removeListener: jest.fn(), // Deprecated
+          addEventListener: jest.fn(),
+          removeEventListener: jest.fn(),
+          dispatchEvent: jest.fn(),
+        }))
+      });
+    });
+
+    it('returns a ref', async () => {
+        const match = media('(hover)')
+
+        expect(isRef(match)).toBe(true)
+    })
+
+})
```

Comment:
does it make sense to test that when the passed in `query` changes, the event listener is updated? Basically testing the callback of the watcher?

Reply:
Same problem as the watcher. Its private and I not able to actually test that it happens